docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,368 @@
+      - [HelmChartProxy CRD](#helmchartproxy-crd)
+        - [Spec:](#spec)
+        - [Status:](#status)
+      - [HelmReleaseProxy CRD](#helmreleaseproxy-crd)
+        - [Spec:](#spec-1)
+        - [Status:](#status-1)
+      - [Controller Design](#controller-design)
+      - [Considerations about the CRD design](#considerations-about-the-crd-design)
+  - [Cluster API enhancements](#cluster-api-enhancements)
+    - [clusterctl](#clusterctl)
+  - [Alternatives](#alternatives)
+    - [Why not CRS](#why-not-crs)
+
+## Glosssary 
+
+**Package management tool:** a tool ultimately responsible[^1] for installing Kubernetes applications, e.g. [Helm](https://helm.sh/) or [carvel-kapp](https://carvel.dev/kapp/).
+
+**Add-on:** an application that extends the functionality of Kubernetes.
+
+**Cluster add-on:** an add-on that is essential to the proper function of the Cluster like e.g. networking (CNI) or integration with the underlying infrastructure provider (CPI, CSI). In some cases the lifecycle of Cluster add-on is strictly linked to the Cluster lifecycle (e.g. CPI must be upgraded in lock step with the Cluster upgrade).
+
+## Summary
+
+Cluster API manages the lifecycle of Kubernetes Clusters; users rely on their own package management tool of choice for Kubernetes application management (e.g Helm, kapp etc.).
+
+This proposal aims to connect the dots between the above two elements, by providing a solution for orchestrating the lifecycle of a subset of Kubernetes applications, the Cluster add-ons, in coordination with the Cluster lifecycle.
+
+While building this solution, we foresee at least three iterations:
+
+Foundation (the one described in this document), with the goal to provide a minimal viable solution for this space, integration with the package management tool of choice - Helm will be used as example -, exploring and addressing basic concern of orchestration like e.g. identifying a single source of truth for add-configuration and reconciliation mechanism to ensure the expected configuration is applied;
+
+Advanced orchestration. As soon as the foundation for addon orchestration will be in place, we will start exploring and addressing advanced orchestration scenarios, like e.g. managing add-on to be updated before the cluster upgrade starts; This work will leverage the lifecycle hooks introduced in Cluster API v1.2.
+
+Add-on dependencies. As a final step of this work, we are going to explore and manage use cases where there are dependencies between add-ons, and eventually introduce capabilities to coordinate add-on management accordingly (e.g upgrade addons in a certain order).
+
+[^1]: Package management tools usually have broader scope that goes over and behind the act of installing a Kuberentes application, like e.g. defining formats for packaging Kubernetes applications, defining repository/marketplaces where users can find libraries of existing applications, templating systems for customizing applications etc.
```

Comment:
formatting looks weird, should this be at the bottom of the doc or in the previous paragraph?